### PR TITLE
HDDS-3597. using protobuf maven plugin instead of the legacy protoc executable file

### PR DESCRIPTION
**Pull request by maobaolong**
## What changes were proposed in this pull request?

With this PR, we united using protobuf-maven-plugin to generate java file from proto file, we never have to install protoc executable program. I think this make new contributor easy to build our repository successfully.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3597

## How was this patch tested?

CI tests

### CODE REVIEW

**Comment by maobaolong** [comment]
@elek Would you like to take a look at this PR？This is a improvement what i aways want to do.

File: hadoop-hdds/common/pom.xml [hunk 1]
```diff
@@ -207,32 +207,51 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
...
-          <protoSourceRoot>${basedir}/src/main/proto/</protoSourceRoot>
-          <includes>
-            <include>DatanodeContainerProtocol.proto</include>
-          </includes>
-          <outputDirectory>target/generated-sources/java</outputDirectory>
-          <clearOutputDirectory>false</clearOutputDirectory>
-        </configuration>
         <executions>
           <execution>
             <id>compile-protoc</id>
```
Comment (mukul1987) [inline hadoop-hdds/common/pom.xml:212]: Lets name this compole-protoc-3

File: hadoop-hdds/common/pom.xml [hunk 2]
```diff
@@ -207,32 +207,51 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
...
+              <outputDirectory>target/generated-sources/java</outputDirectory>
+              <clearOutputDirectory>false</clearOutputDirectory>
+              <pluginId>grpc-java</pluginId>
+              <pluginArtifact>
+                io.grpc:protoc-gen-grpc-java:${grpc-compile.version}:exe:${os.detected.classifier}
+              </pluginArtifact>
+            </configuration>
+          </execution>
+          <execution>
+            <id>compile-protoc2</id>
```
Comment (mukul1987) [inline hadoop-hdds/common/pom.xml:236]: and this can be named compile-protoc-2

**Comment by mukul1987** [comment]
@maobaolong, the changes look good to me
can we also remove protoc.path in https://github.com/apache/hadoop-ozone/blob/53e7018dcc24d2fae43bedb71147455b94cf76ba/pom.xml#L155

File: hadoop-hdds/common/pom.xml (hunk 2 above)
Comment (maobaolong) [reply hadoop-hdds/common/pom.xml:236]: Done

File: hadoop-hdds/common/pom.xml (hunk 1 above)
Comment (maobaolong) [reply hadoop-hdds/common/pom.xml:212]: Done

**Comment by maobaolong** [comment]
@mukul1987 Thanks for your comments, i have addressed all your comments, PTAL.

**Comment by elek** [comment]
Just moving my comments from the duplicated PR:

> Wow, very good idea. Thanks to address this issue. Will test it soon.

> Can you please remove the related parts from `CONTRIBUTING.md` in the mean time? (protoc 2.5 related parts can be removed)

**Comment by elek** [comment]
>  Will test it soon.

Tested and worked well on Linux. Bye bye `protoc 2.5`...

I am +1 when the `CONTRIBUTION.md` is updated...

**Comment by maobaolong** [comment]
@elek Updated the `CONTRIBUTING.md` now, Bye bye `protoc 2.5`, thanks for your review.

**Comment by umamaheswararao** [comment]
Thanks @maobaolong and @vinayakumarb for doing this. Big +1 
:-)

**Comment by elek** [comment]
Merged. Thanks to improvement @maobaolong, and the review all the others.

**Comment by maobaolong** [comment]
@elek I've resolved the conflicts, PTAL

**Comment by maobaolong** [comment]
Hi @elek Is there something should i do else? Would you like to help me to merge this PR?

**Comment by adoroszlai** [comment]
> Hi @elek Is there something should i do else? Would you like to help me to merge this PR?

Actually, it has been committed (I guess accidentally) to `master` in cf6631eeb0bd638fbb9ef1db5d263024d6f87a85.

**Comment by maobaolong** [comment]
@adoroszlai Oh, really? its surprising me.

**Comment by mukul1987** [comment]
Closing the pull request as this is already merged.

**Comment by elek** [comment]
> Actually, it has been committed (I guess accidentally) to master in cf6631e.

It's merged, but not accidentally. (See my comments above). Usually I merge it locally and push it. It makes the signing possible but an additional comment should be added to the commit message to auto-close the PR.

Recently I started to use the merge button as anybody else. Signature is not possible, but the view of PR will be consistent with the view of the other PRs.

**Comment by adoroszlai** [comment]
> It's merged, but not accidentally. (See my comments above).

Sorry, I hadn't noticed that comment.  I guessed it was accidental because it didn't look like your previous "closes ..." commits (and I thought you had a script for it).

**Pull request closed without merging**